### ISSUE TITLE
Sphinx is broken because of a circular dependency in DH

### ISSUE THREAD

**Issue by chipkent** [deephaven-examples/deephaven-ib#43]
Sphinx is failing to generate documentation because of a circular dependency in DH.

The following code is from `dh_sphinx.py`.  Here you will notice that `setup_sphinx_environment` is a function to setup and configure the sphinx environment as well as the jpy jvm.

```
import os
import pkgutil
import shutil
import sys
from pathlib import Path

import jpy
from deephaven.start_jvm import start_jvm

def setup_sphinx_environment():
    # add the deephaven-ib path
    new_python_path = Path(os.path.realpath(__file__)).parents[2].joinpath("src")
    sys.path.append(str(new_python_path))

    # start the jvm so that deephaven can be loaded
    if not jpy.has_jvm():
        os.environ['JAVA_VERSION'] = '11'
        start_jvm(devroot="/tmp", workspace="/tmp", propfile='dh-defaults.prop',
                  java_home=os.environ.get('JDK_HOME', None),
                  jvm_classpath="/opt/deephaven/server/lib/*", skip_default_classpath=True)
```

As of DH `v0.12.0`, the following exception results.  Note that here the `from deephaven.start_jvm import start_jvm` import is triggering `jpy` calls, but these calls are before the `jpy` JVM has been configured.

```
Traceback (most recent call last):
  File "/opt/deephaven-venv/lib/python3.7/site-packages/sphinx/config.py", line 332, in eval_config_file
    exec(code, namespace)
  File "/build/deephaven-ib/sphinx/source/conf.py", line 79, in <module>
    import dh_sphinx
  File "/build/deephaven-ib/sphinx/lib/dh_sphinx.py", line 8, in <module>
    from deephaven.start_jvm import start_jvm
  File "/opt/deephaven-venv/lib/python3.7/site-packages/deephaven/__init__.py", line 12, in <module>
    from .table import SortDirection, AsOfMatchRule
  File "/opt/deephaven-venv/lib/python3.7/site-packages/deephaven/table.py", line 12, in <module>
    from deephaven import DHError, dtypes
  File "/opt/deephaven-venv/lib/python3.7/site-packages/deephaven/dtypes.py", line 17, in <module>
    _JQstType = jpy.get_type("io.deephaven.qst.type.Type")
RuntimeError: jpy: No JVM available.
```

**Comment by chipkent** [comment]
We will wait on deephaven-as-a-library to fix this.

**Comment by chipkent** [comment]
Resolved by #63
